tests/unit/fem/test_lin_interp.cpp

```diff
@@ -59,164 +106,851 @@ double FdotG3(const Vector & x)
+         SECTION("Mapping to L2 (INTEGRAL)")
+         {
+            L2_FECollection    fec_l2(order_l2, dim,
+                                      BasisType::GaussLegendre,
+                                      FiniteElement::INTEGRAL);
+            FiniteElementSpace fespace_l2(&mesh, &fec_l2);
 
-         DiscreteLinearOperator OpF1(&fespace_h1,&fespace_ndp);
-         OpF1.AddDomainInterpolator(new VectorScalarProductInterpolator(FCoef));
-         OpF1.Assemble();
+            GridFunction dF1(&fespace_l2);
 
-         GridFunction Fg1(&fespace_ndp);
-         OpF1.Mult(g0,Fg1);
+            DiscreteLinearOperator Op(&fespace_nd,&fespace_l2);
+            Op.AddDomainInterpolator(new CurlInterpolator());
+            Op.Assemble();
 
-         REQUIRE( Fg1.ComputeL2Error(FgCoef) < tol );
+            Op.Mult(F1,dF1);
+
+            REQUIRE( dF1.ComputeL2Error(CurlFCoef) < tol );
+         }
+      }
+      SECTION("Operators on HDiv for element type " + std::to_string(type))
+      {
+         RT_FECollection    fec_rt(order_rt, dim);
+         FiniteElementSpace fespace_rt(&mesh, &fec_rt);
+
+         GridFunction F2(&fespace_rt);
+         F2.ProjectCoefficient(FCoef);
+
+         SECTION("Mapping to L2")
+         {
+            L2_FECollection    fec_l2(order_l2, dim);
+            FiniteElementSpace fespace_l2(&mesh, &fec_l2);
+
+            GridFunction dF2(&fespace_l2);
+
+            DiscreteLinearOperator Op(&fespace_rt,&fespace_l2);
+            Op.AddDomainInterpolator(new DivergenceInterpolator());
+            Op.Assemble();
+
+            Op.Mult(F2,dF2);
+
+            REQUIRE( dF2.ComputeL2Error(DivFCoef) < tol );
+         }
+         SECTION("Mapping to L2 (INTEGRAL)")
+         {
+            L2_FECollection    fec_l2(order_l2, dim,
+                                      BasisType::GaussLegendre,
+                                      FiniteElement::INTEGRAL);
+            FiniteElementSpace fespace_l2(&mesh, &fec_l2);
+
+            GridFunction dF2(&fespace_l2);
+
+            DiscreteLinearOperator Op(&fespace_rt,&fespace_l2);
+            Op.AddDomainInterpolator(new DivergenceInterpolator());
+            Op.Assemble();
+
+            Op.Mult(F2,dF2);
+
+            REQUIRE( dF2.ComputeL2Error(DivFCoef) < tol );
+         }
       }
    }
-   SECTION("Operators on HCurl")
+}
+
+TEST_CASE("3D Derivative Linear Interpolators",
+          "[GradientInterpolator]"
+          "[CurlInterpolator]"
+          "[DivergenceInterpolator]")
+{
+   int order_h1 = 1, order_nd = 1, order_rt = 0, order_l2 = 0, n = 3, dim = 3;
+   double tol = 1e-9;
+
+   FunctionCoefficient       fCoef(f3);
+   VectorFunctionCoefficient FCoef(dim, F3);
+
+   VectorFunctionCoefficient GradfCoef(dim, Grad_f3);
+   VectorFunctionCoefficient CurlFCoef(dim, CurlF3);
+   FunctionCoefficient       DivFCoef(DivF3);
+
+   for (int type = (int)Element::TETRAHEDRON;
+        type <= (int)Element::HEXAHEDRON; type++)
```

Comment:
Could the 1D-3D cases be collapsed into this loop like the other test?

Reply:
Yes, it could.